### ISSUE TITLE
web-angular: rework template and add basic component tests

### ISSUE THREAD

**Issue by chalin** [google/stagehand#396]
Part of #390.

**Comment by chalin** [comment]
The current template allows the user to type her/his name and get a "Hello" message inside a modal dialog. Part of the issue is that we aren't able to write component tests for modal dialogs quite yet.

@kwalrath @filiph - these are the changes that I propose doing:
1. Drop `hello_dialog`: instead, display the greeting in a "Greeting" tab (this can be tested). Looking like this
> ---
> ![screen shot 2017-05-04 at 7 51 33 pm](https://cloud.githubusercontent.com/assets/4140793/25729394/1c79cda2-3103-11e7-877d-9ea455d89f64.png)

> ---
> ![screen shot 2017-05-04 at 7 43 09 pm](https://cloud.githubusercontent.com/assets/4140793/25729408/2deb5c4a-3103-11e7-9143-4e9a34fce1b0.png)

2. Have a second tab named "Hall of Fame" that will be initially empty. This second tab will be populated by code which is added in the second part of the [Get Started](https://webdev.dartlang.org/guides/get-started) page; i.e. from this
> ![screen shot 2017-05-04 at 7 43 55 pm](https://cloud.githubusercontent.com/assets/4140793/25729460/67adc1a2-3103-11e7-8acc-16978e276498.png)

To this:
> ![screen shot 2017-05-04 at 7 42 55 pm](https://cloud.githubusercontent.com/assets/4140793/25729468/76fe6bb6-3103-11e7-956f-2ba951da1640.png)

Clicking a superhero name shows a "cheers!" dialog (which is the old `hello_dialog` repurposed). (We don't include the tests for this part of the code, so using a modal dialog is ok).

WDYT? Suggestions are welcome.

**Comment by chalin** [comment]
@kwalrath @filiph - FYI, here is a snapshot of the revised app:
> ![screen shot 2017-05-08 at 5 12 22 pm](https://cloud.githubusercontent.com/assets/4140793/25825605/8bca220e-3411-11e7-97ed-11aa974bad84.png)



I'm looking at the tests now. Once that is done, I'll submit a new PR with the revised code.

**Comment by filiph** [comment]
Thanks! This looks amazing! 

Where are we on the LOC front? From the screenshot it looks like a full-featured TODO app, which is awesome, but only if it's really just a few lines of code (in the ballpark of the current template).

If it's more than that, let's use your code as a medium article / sample / etc., because this is _exactly_ what we need right now — but lets shave the code in the template to a bare minimum.

**Comment by chalin** [comment]
@filiph - the code (w/o the updated tests) is in #401. I'm stepping out, but will be available for discussion when I get back.
